lib/KTextTruncator.vue

```diff
@@ -0,0 +1,151 @@
+<template>
+
+  <!--
+    Text is wrapped in two `spans`s to allow parent components adding
+    padding style directly on `<TextTruncatorCss>` component no matter
+    of what truncating technique is used. Otherwise adding padding directly
+    would break when using technique (B) since text that should be truncated
+    would show in padding area.
+
+    Attributes are inherited by the inner `span` to emulate the same behavior
+    like if only one element would wrap the text to allow attributes be applied
+    as close as possible to the text element.
+
+    Some width information need to be provided to `<span>s` to allow `text-overflow`
+    calculate properly when ellipsis should be added.
+  -->
+  <span :style="{ display: 'inline-block', maxWidth: '100%' }">
+    <span
+      v-bind="$attrs"
+      :style="{ display: 'inline-block', maxWidth: '100%' }"
+      :class="$computedClass(truncate)"
+    >
+      {{ text }}
+    </span>
+  </span>
+
+</template>
+
+
+<script>
+
+  /**
+   * Truncates text to a certain number of lines
+   * and adds an ellipsis character "…"
+   *
+   * Internet Explorer note:
+   * Depending on length of words of the text, there might
+   * be a gap between the last visible word and "…"
+   */
+  export default {
+    name: 'KTextTruncator',
+    inheritAttrs: false,
+    props: {
+      /**
+       * Text to be truncated
+       */
+      text: {
+        type: String,
+        required: true,
+      },
+      /**
+       * Maximum number of lines to be shown
+       */
+      maxLines: {
+        type: Number,
+        required: false,
+        default: 1,
+      },
+      /**
+       * Text line height in rem.
+       * Used only for Internet Explorer fallback.
+       */
+      lineHeightIE: {
+        type: Number,
+        required: false,
+        default: 1.4,
+      },
+    },
+    computed: {
+      truncate() {
+        const nuxtClientSideRendering = process.client;
+
+        /*
+          (A)
+          For one line, use standard ellipsis text overflow
+          that works well for such scenario
+        */
+        if (this.maxLines === 1) {
+          return {
+            overflow: 'hidden',
+            textOverflow: 'ellipsis',
+            whiteSpace: 'nowrap',
+          };
+        }
+
+        // 54 is random number only to be able to define `supports` test condition
+        if (
+          nuxtClientSideRendering &&
```

Comment:
@muditchoudhary Thanks for the naming change and with apology I missed this during the first round of review - what was actually the reason for adding the `process.client` check at this particular place? 

Now when I look at all conditions in `truncate` method as a whole, originally it said something like:

if one line - use ellipsis
if multiple lines - if a browser supports `webkit-line-clamp` then use it, else use IE fallback

Now with your addition it says something along the lines

if one line - use ellipsis
if multiple lines - if a browser supports `webkit-line-clamp` **and is not server side rendering** then use it, else use IE fallback

which doesn't seem to make sense high-level, right?

Reply:
Ah was it about accessing the `window`?

Reply:
If so, I think we could rather place this check at the beginning of the `truncate` method just to make it a bit more clear. I will wait for your reply though to be sure what was the motivation for having it here.

Reply:
@muditchoudhary If that's the case, I'd suggest to rather put this at the very beginning of `truncate` method:

```
const nuxtServerSideRendering = process && process.server;
if (nuxtServerSideRendering) {
  return;
}
```

just to make the placement of that condition a bit more clear. This is similar to other places in KDS.

Reply:
It's rather a minor note, nice to have. Nothing blocking.